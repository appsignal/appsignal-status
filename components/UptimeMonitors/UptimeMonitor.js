import PropTypes from "prop-types";
import Tippy from "@tippyjs/react";

const Bar = () => {
  return (
    <Tippy
      content={
        <div className="text-center text-sm">
          <p className="text-gray-200">July 19th</p>
          <p>No outage</p>
        </div>
      }
    >
      <div className="h-8 flex-grow bg-green-500 rounded" />
    </Tippy>
  );
};

const UptimeMonitor = ({ uptimeMonitor }) => {
  return (
    <div className="px-6 py-5 space-y-3">
      <div className="sm:flex justify-between">
        <h2 className="c_h-heading">{uptimeMonitor.title}</h2>
        <p className="mt-1 sm:mt-0 text-gray-700">
          Monitoring from 4 locations
        </p>
      </div>
      <div className="flex space-x-1">
        {[...Array(30)].map((e, index) => (
          <Bar key={index} />
        ))}
      </div>
    </div>
  );
};

UptimeMonitor.propTypes = {
  uptimeMonitor: PropTypes.object.isRequired,
};

export default UptimeMonitor;
