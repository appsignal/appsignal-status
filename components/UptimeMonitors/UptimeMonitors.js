import PropTypes from "prop-types";
import UptimeMonitor from "../UptimeMonitor";

const UptimeMonitors = ({ statusPage }) => {
  const renderUptimeMonitors = () => {
    if (statusPage.uptime_monitors.length === 0) {
      return (
        <div className="max-w-lg mx-auto bg-white shadow-sm rounded px-6 py-5">
          <p className="text-gray-700 italic">No uptime monitors added yet.</p>
        </div>
      );
    }

    return statusPage.uptime_monitors.map((monitor) => (
      <UptimeMonitor
        key={monitor.url}
        uptimeMonitor={monitor}
        threshold={statusPage.threshold}
      />
    ));
  };
  return (
    <section className="mt-16 mb-20">
      <div className="container">
        <div className="max-w-lg mx-auto bg-white shadow-sm rounded divide-y divide-gray-200">
          {renderUptimeMonitors()}
        </div>
      </div>
    </section>
  );
};

UptimeMonitors.propTypes = {
  statusPage: PropTypes.shape({
    uptime_monitors: PropTypes.array.isRequired,
    threshold: PropTypes.number,
  }).isRequired,
};

export default UptimeMonitors;
