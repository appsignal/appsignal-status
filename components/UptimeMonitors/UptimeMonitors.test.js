import { render, screen, waitFor } from "@testing-library/react";

import UptimeMonitors from "./UptimeMonitors";

import statusPageMock from "../../mocks/status_pages/appsignal.json";

const build = (props = {}) => {
  return render(<UptimeMonitors statusPage={statusPageMock} {...props} />);
};

describe("UptimeMonitors", () => {
  test("renders without problems", () => {
    const { container } = build();
    expect(container).toMatchSnapshot();
  });

  test("should render an empty state when there are no monitors", () => {
    const statusPage = Object.assign({}, statusPageMock);
    statusPage.uptime_monitors = [];

    build({ statusPage });

    expect(
      screen.getByText("No uptime monitors added yet.")
    ).toBeInTheDocument();
  });

  test("renders multiple uptime monitors", async () => {
    build();

    await waitFor(() => {
      expect(screen.getByText("homepage")).toBeInTheDocument();
    });

    await waitFor(() => {
      expect(screen.getByText("blog")).toBeInTheDocument();
    });

    await waitFor(() => {
      expect(screen.getByText("Always Down")).toBeInTheDocument();
    });
  });

  test("should render uptime monitors sorted by title", () => {
    const statusPage = {...statusPageMock};

    build({ statusPage });

    const updates = screen.getAllByTestId("UptimeMonitor");

    expect(updates[0].querySelector("h2").innerHTML).toContain(
      statusPageMock.uptime_monitors[0].title
    );

    expect(updates[1].querySelector("h2").innerHTML).toContain(
      statusPageMock.uptime_monitors[1].title
    );

    expect(updates[2].querySelector("h2").innerHTML).toContain(
      statusPageMock.uptime_monitors[2].title
    );
  });
});
