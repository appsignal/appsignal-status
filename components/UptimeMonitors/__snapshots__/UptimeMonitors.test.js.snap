// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UptimeMonitors renders without problems 1`] = `
<div>
  <section
    class="mt-16 mb-20"
  >
    <div
      class="container"
    >
      <div
        class="max-w-lg mx-auto bg-white shadow-sm rounded divide-y divide-gray-200"
      >
        <div
          class="px-6 py-5 space-y-3"
        >
          <div
            class="sm:flex justify-between"
          >
            <h2>
              <button
                class="c_h-heading focus:outline-none"
              >
                homepage
              </button>
            </h2>
            <p
              class="mt-1 sm:mt-0 text-gray-700"
            >
              Monitoring from 
              4
               
              locations
            </p>
          </div>
          <div
            class="flex space-x-1"
          >
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
          </div>
        </div>
        <div
          class="px-6 py-5 space-y-3"
        >
          <div
            class="sm:flex justify-between"
          >
            <h2>
              <button
                class="c_h-heading focus:outline-none"
              >
                blog
              </button>
            </h2>
            <p
              class="mt-1 sm:mt-0 text-gray-700"
            >
              Monitoring from 
              4
               
              locations
            </p>
          </div>
          <div
            class="flex space-x-1"
          >
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
          </div>
        </div>
        <div
          class="px-6 py-5 space-y-3"
        >
          <div
            class="sm:flex justify-between"
          >
            <h2>
              <button
                class="c_h-heading focus:outline-none"
              >
                Always Down
              </button>
            </h2>
            <p
              class="mt-1 sm:mt-0 text-gray-700"
            >
              Monitoring from 
              4
               
              locations
            </p>
          </div>
          <div
            class="flex space-x-1"
          >
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
            <div
              class="h-8 flex-grow rounded bg-gray-200"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
